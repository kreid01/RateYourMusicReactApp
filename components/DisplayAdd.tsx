import React, { useState } from "react";
import { Animated, View, Text } from "react-native";

export const DisplayAdd = ({ releaseTitle }: { releaseTitle: string }) => {
  const progressBar = useState(new Animated.Value(0))[0];

  useState(() => {
    Animated.timing(progressBar, {
      toValue: 372,
      duration: 2250,
      useNativeDriver: false,
    }).start();
  });

  return (
    <View className=" bg-gray-700 h-8 w-[90vw]">
      <Animated.View
        style={{
          width: progressBar,
        }}
        className="absolute border-2 border-red-500"
      />
      <Text className="text-white mx-auto my-auto">
        {releaseTitle} added to playlist{" "}
      </Text>
    </View>
  );
};
