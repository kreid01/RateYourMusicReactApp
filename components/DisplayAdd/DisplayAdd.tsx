import React, { useState } from "react";
import { Animated, View, Text } from "react-native";

export const DisplayAdd = ({ releaseTitle }: { releaseTitle: string }) => {
  const progressBar = useState(new Animated.Value(0))[0];

  useState(() => {
    Animated.timing(progressBar, {
      toValue: 372,
      duration: 4529,
      useNativeDriver: false,
    }).start();
  });

  return (
    <View className=" bg-gray-700 h-8 w-[90vw]">
      <Animated.View
        style={{
          width: progressBar,
        }}
        className="absolute border-2 border-blue-500"
      />
      <Text testID="playlist-add" className="text-white pt-1 mx-auto my-auto">
        {releaseTitle} added to playlist{" "}
      </Text>
    </View>
  );
};
