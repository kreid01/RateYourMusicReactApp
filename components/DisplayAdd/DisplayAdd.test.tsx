import React from "react";
import {
  screen,
  render,
  cleanup,
  fireEvent,
} from "@testing-library/react-native";
import "@testing-library/jest-dom";
import { NativeBaseProvider } from "native-base";
import renderer, { act } from "react-test-renderer";
import { expect } from "@jest/globals";
import { DisplayAdd } from "./DisplayAdd";

const handleEdit = jest.fn();

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

afterEach(cleanup);

it("matched snapshot", async () => {
  const tree = renderer
    .create(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <DisplayAdd releaseTitle="test" />
      </NativeBaseProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("hides component if cancel button is clicked", async () => {
  render(
    <NativeBaseProvider initialWindowMetrics={inset}>
      <DisplayAdd releaseTitle="test" />
    </NativeBaseProvider>
  );
  screen.getByTestId("playlist-add");
});
