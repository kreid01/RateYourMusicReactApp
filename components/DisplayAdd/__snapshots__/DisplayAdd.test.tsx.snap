// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    className=" bg-gray-700 h-8 w-[90vw]"
  >
    <View
      className="absolute border-2 border-blue-500"
      collapsable={false}
      style={
        {
          "width": 0,
        }
      }
    />
    <Text
      className="text-white pt-1 mx-auto my-auto"
      testID="playlist-add"
    >
      test
       added to playlist
       
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
